ksqldb-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -889,10 +889,21 @@ public Node visitNullPredicate(final SqlBaseParser.NullPredicateContext context)
 
     @Override
     public Node visitLike(final SqlBaseParser.LikeContext context) {
+
+      final Optional<String> escape = Optional.ofNullable(context.escape)
+          .map(Token::getText)
+          .map(s -> ParserUtil.unquote(s, "'"));
+      escape.ifPresent(s -> {
+        if (s.length() != 1) {
+          throw new IllegalStateException("Expected single character escape but got: " + s);
```

Comment:
IllegalStateException?

The syntax allows the escape to be a multi-char string, so this is the error the users going to see if they enter one, right?

I'd of thought this should be :
```suggestion
          throw new KsqlException(loc.asPrefix() + "Expected single character escape but got: " + s);
```

Where `loc` is the `NodeLocation`.

Reply:
ah good catch, originally I had it in the syntax as just a single char (`.`) but then i changed it to `STRING` to include the quotes and forgot to change this!